pkg/pool/pool_test.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pool
+
+import (
+	"errors"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+)
+
+func TestParallelismNoErrors(t *testing.T) {
+	tests := []struct {
+		name string
+		size int
+		work int
+	}{{
+		name: "single threaded",
+		size: 1,
+		work: 3,
+	}, {
+		name: "three workers",
+		size: 3,
+		work: 10,
+	}, {
+		name: "ten workers",
+		size: 10,
+		work: 100,
+	}}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// m guards max
```

Comment:
1. t.Parallel() for increased fun? (also will require tc := tc above `Run`).
2. var (...)
3.
```suggestion
			// m guards max.
```

Reply:
This is fast enough (even with sleeps) that `t.Parallel` doesn't get use much, so I'd rather keep things simple and avoid the `tc := tc` wart. :(